### PR TITLE
Replace complex to channels with Keras native

### LINKED ISSUE #291: Complex to channels and back now in Keras
We currently have [`zea.func.complex_to_channels`](https://zea.readthedocs.io/en/latest/_autosummary/zea.func.html#zea.func.channels_to_complex) as well as [`zea.func.channels_to_complex`](https://zea.readthedocs.io/en/latest/_autosummary/zea.func.html#zea.func.channels_to_complex). These might have become obsolete with the introduction of Keras [`keras.ops.view_as_complex`](https://github.com/keras-team/keras/blob/v3.13.2/keras/src/ops/math.py#L1064) added in 3.10.0, see https://github.com/keras-team/keras/pull/21221

### PR DESCRIPTION
**Pull request by tristan-deep**
See #291 

## Summary by CodeRabbit

- **Improvements**
  - Enhanced complex IQ handling across filtering, beamforming, demodulation, and upmixing using view-based complex/real conversions.
  - Preserved the requested channel-axis placement when converting between complex tensors and real/imaginary channels.
  - Complex-channel FIR filtering now preserves shape, matches equivalent independent real-channel filtering, and raises an error for invalid complex-channel axis layouts.
- **Compatibility**
  - Kept legacy conversion helpers as deprecated wrappers that forward to the standard Keras ops.
- **Tests**
  - Added coverage for complex-to-channels-to-complex round-trips and complex-channel FIR behavior across backends.

### CODE REVIEW

File: zea/ops/ultrasound.py [hunk 1]
```diff
@@ -733,7 +731,7 @@ def _validate_and_unpack_passband(selected_passband):
 class ChannelsToComplex(Operation):
```
Comment (wesselvannierop) [inline zea/ops/ultrasound.py:732]: Do we want to rename these operations maybe?

Comment (tristan-deep) [reply zea/ops/ultrasound.py:732]: If we rename we might as well remove? They are already in keras ops and by virtue in zea right?

Comment (tristan-deep) [reply zea/ops/ultrasound.py:732]: I have now decorated them with `@deprecated`, pointing these (now redundant classes) to `zea.ops.keras_ops.ViewAsComplex` and `zea.ops.keras_ops.ViewAsReal`, respectively. We can remove them in future releases.

**Review by wesselvannierop — APPROVED** [review]
